api/rest/client/lbclient.go

```diff
@@ -13,27 +12,44 @@ import (
 	peer "github.com/libp2p/go-libp2p-peer"
 )
 
+// Strategy values
+const (
+	Random Strategy = iota
+	RoundRobin
+)
+
+// Strategy represents load balancing strategy
+type Strategy int
+
 // loadBalancingClient a client to interact with IPFS Cluster APIs.
 // It balances the load by distributing it among peers.
 type loadBalancingClient struct {
-	clients           []*Client
-	shedulingStrategy string
+	clients  []*Client
+	strategy Strategy
+	retries  int
+	length   int
 }
 
-func (lc *loadBalancingClient) whichClient() (Client, error) {
-	if lc.shedulingStrategy == "random" {
-		r := rand.New(rand.NewSource(time.Now().UnixNano()))
-		n := r.Intn(len(lc.clients))
+var clientIndex int
 
-		return *lc.clients[n], nil
+func (lc *loadBalancingClient) whichClient() Client {
+	switch lc.strategy {
+	case Random:
+		return *lc.clients[rand.Intn(len(lc.clients))]
+	case RoundRobin:
+		clientIndex++
+		if clientIndex == lc.length {
+			clientIndex = 0
+		}
```

Comment:
`clientIndex` is racy if multiple client requests are made by the peer at the same time. Should I assume that it would happen?

Reply:
Yes